src/utils/graph/constraints.js

```diff
@@ -0,0 +1,99 @@
+import { distance1d, greaterOrEqual, equalTo, subtract } from './common';
+
+/**
+ * Constraint in Y for separating rows
+ */
+export const rowConstraint = {
```

Comment:
It would be really helpful to add a comment to explain what each field under the Constraint object is used for and which parts of the graph it affects - would definitely be helpful to provide clarity for future development work.

Reply:
Ah yes the docs for these fields are on the `solve` function in `solver.js`, I'll add a comment in this file to make note of that ten